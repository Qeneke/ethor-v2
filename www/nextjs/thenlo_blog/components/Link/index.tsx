import { useRouter } from "next/router";
import Link, { LinkProps } from "next/link";
import React, { ReactElement, SFC } from "react";

interface Props {
  children: ReactElement;
  activeClassName?: string;
}

const EnhancedLink: SFC<Props & LinkProps> = ({
  children,
  ...props
}): JSX.Element => {
  const child = React.Children.only(children);
  const router = useRouter();
  const { href, activeClassName } = props;

  let className = child.props.className || null;
  if (
    router &&
    router.pathname ===
      (typeof href !== "string" ? href.pathname : href.split("?")[0]) &&
    activeClassName
  ) {
    className = `${className || ""} ${activeClassName}`.trim();
  }

  // eslint-disable-next-line no-param-reassign
  delete props.activeClassName;

  return <Link {...props}>{React.cloneElement(child, { className })}</Link>;
};

export default EnhancedLink;
