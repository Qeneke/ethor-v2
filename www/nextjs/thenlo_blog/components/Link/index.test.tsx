/* eslint-env jest */
import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import Link from ".";

describe("www:nextjs:thenlo:components:Link", (): void => {
  it("renders the link with correct a tag", (): void => {
    const wrapper = shallow(
      <Link href="/user">
        <a>To User</a>
      </Link>
    );
    expect(wrapper.find("a").text()).toEqual("To User");
  });
  it("render link with snapshot", (): void => {
    const component = renderer.create(
      <Link href="/user">
        <a>To User</a>
      </Link>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
