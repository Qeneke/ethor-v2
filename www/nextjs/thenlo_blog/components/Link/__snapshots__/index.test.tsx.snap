// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`www:nextjs:thenlo:components:Link render link with snapshot 1`] = `
<a
  className={null}
  href="/user"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  To User
</a>
`;
